src/extensionProvider.ts

```diff
@@ -93,65 +93,39 @@ export class ExtensionProvider implements IDappProvider {
     return !!this.account;
   }
 
-  async sendTransaction(transaction: ITransaction): Promise<ISignedTransaction> {
-    const txResponse = await this.startBgrMsgChannel("sendTransactions", {
-      from: this.account.address,
-      transactions: [transaction.toPlainObject()],
-    });
-
-    return this.getTransactionFactory().fromPlainObject(txResponse[0]);
+  async signTransaction<T extends ITransaction>(transaction: T): Promise<T> {
+    let response = await this.signTransactions([transaction]);
+    return response[0];
   }
 
-  async signTransaction(transaction: ITransaction): Promise<ISignedTransaction> {
-    const txResponse = await this.startBgrMsgChannel("signTransactions", {
+  async signTransactions<T extends ITransaction>(transactions: Array<T>): Promise<Array<T>> {
+    let extensionResponse = await this.startBgrMsgChannel("signTransactions", {
       from: this.account.address,
-      transactions: [transaction.toPlainObject()],
+      transactions: transactions.map(transaction => transaction.toPlainObject()),
     });
-    return this.getTransactionFactory().fromPlainObject(txResponse[0]);
-  }
 
-  async signTransactions(
-    transactions: Array<ITransaction>
-  ): Promise<Array<ITransaction>> {
-    transactions = transactions.map((transaction) =>
-      transaction.toPlainObject()
-    );
-    let txResponse = await this.startBgrMsgChannel("signTransactions", {
-      from: this.account.address,
-      transactions: transactions,
-    });
     try {
-      // TODO: Find out whether transaction.applySignature() is better suited here,
-      // It seems that the extension only sets "sender" and "signature" 
-      // (version, nonce, chainID, gasLimit etc. do not seem to ever be overridden).
-      // If so, we don't need factories here.
-      txResponse = txResponse.map((transaction: any) =>
-        this.getTransactionFactory().fromPlainObject(transaction)
-      );
-    } catch (error) {
-      throw new Error("Transaction canceled.");
+      for (let i = 0; i < transactions.length; i++) {
```

Comment:
I wonder if maybe we should a .map here and return the transactions with applied signature, will give you a mental image that the returned transaction has its signature applied

Reply:
On one hand, it would be better. On the other hand, as the code is now, it's more explicit that we actually return same transactions passed as input. 

And now I realize - since we always return the same objects passed as input, with the exception of the web-wallet-provider (which returns nothing in the homologous functions), we can remove the output parameters in all signing providers. How would this be? @StanislavSava @arhtudormorar @radumojic @andreigiura.